delfin/task_manager/tasks/resources.py

```diff
@@ -53,6 +54,11 @@ def _set_synced_after(func, *args, **kwargs):
                     storage['sync_status'] -= sync_result
                     db.storage_update(self.context, self.storage_id, storage)
 
+                    if storage['sync_status'] == 0:
```

Comment:
I think we need to make this before updating the sync status in DB.. because once you update the sync status in DB another sync could come..

Reply:
Updated